lib/misc.js

```diff
@@ -65,35 +65,51 @@ exports.generateId = function (fields) {
   return base32hex.encodeBuffer(hasher.digest(), { paddingChar: '-' });
 };
 
-exports.encryptArrayToFile = function (dataArray, filePath, env, serverSecret) {
-  var iv = `${env}-environment`.substring(0, 16);
-  var key = serverSecret.substring(0, 32);
-  var algorithm = 'aes-256-cbc';
+var userIdsConfig = function (env, serverSecret) {
+  const key = crypto.createHash('sha256').update(serverSecret).digest('hex');
 
-  var encryptedArray = dataArray.map((item) => {
-    var cipher = crypto.createCipheriv(algorithm, key, iv);
-    let encrypted = cipher.update(item, 'utf8', 'hex');
-    encrypted += cipher.final('hex');
-    return encrypted;
-  });
-  fs.writeFileSync(filePath, JSON.stringify(encryptedArray));
+  return {
+    encryptedFilePath: `${__dirname}/platform_users/${env}.json.enc`,
+    ivBlockSize: 16,
+    key: key.substring(0, 32),
+    algorithm: 'aes-256-cbc',
+  };
 };
 
-exports.decryptArrayFromFile = function (filePath, env, serverSecret) {
-  if (!fs.existsSync(filePath)) {
-    throw new Error(`Missing required file ${filePath}`);
+// Function to encrypt the contents of a JSON file
+exports.encryptUserIds = function (inputFilePath, env, serverSecret) {
+  const config = userIdsConfig(env, serverSecret);
+
+  const data = fs.readFileSync(inputFilePath, 'utf8');
+  const jsonArray = JSON.parse(data);
+  const dataString = JSON.stringify(jsonArray);
+  const iv = crypto.randomBytes(config.ivBlockSize);
+
+  const cipher = crypto.createCipheriv(config.algorithm, config.key, iv);
+  let encrypted = cipher.update(dataString, 'utf8', 'hex');
```

Comment:
Do you want to encoded it all as hex? Why not just leave it as binary? Though, I suppose hex text is easier for git comparisons.

Reply:
leaving as hex